### PR TITLE
Types for DnD, reuse and connect sidebar components to redux

### PR DESCRIPTION
**Pull request by develohpanda**
As per the title, demo shows DnD working properly after refactoring

### CODE REVIEW

File: packages/insomnia-app/app/ui/components/dropdowns/request-group-actions-dropdown.tsx [hunk 1]
```diff
@@ -131,7 +133,7 @@ class RequestGroupActionsDropdown extends PureComponent<Props, State> {
     } = this.props;
     const { actionPlugins, loadingActions } = this.state;
     return (
-      <Dropdown ref={this._setDropdownRef} onOpen={this.onOpen} {...(other as Record<string, any>)}>
+      <Dropdown ref={this._setDropdownRef} onOpen={this.onOpen} {...other}>
```
Comment (dimitropoulos) [inline packages/insomnia-app/app/ui/components/dropdowns/request-group-actions-dropdown.tsx:136]: yeppppppppperrrrrrdooooooo

File: packages/insomnia-app/app/ui/components/sidebar/sidebar-request-group-row.tsx [hunk 2]
```diff
@@ -1,68 +1,61 @@
...
-
-  _setRequestGroupActionsDropdownRef(n: RequestGroupActionsDropdown) {
-    this._requestGroupActionsDropdown = n;
-  }
-
-  _setExpandTagRef(n: HTMLDivElement) {
-    this._expandTag = n;
-  }
+  private dropdownRef = createRef<UnconnectedRequestGroupActionsDropdown>();
+  private expandTagRef = createRef<HTMLDivElement>();
```
Comment (dimitropoulos) [inline packages/insomnia-app/app/ui/components/sidebar/sidebar-request-group-row.tsx:55]: fantastic.

File: packages/insomnia-app/app/ui/components/wrapper-debug.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ import GrpcRequestPane from './panes/grpc-request-pane';
 import GrpcResponsePane from './panes/grpc-response-pane';
 import RequestPane from './panes/request-pane';
 import ResponsePane from './panes/response-pane';
-import SidebarChildren from './sidebar/sidebar-children';
+import { SidebarChildren } from './sidebar/sidebar-children';
```
Comment (dimitropoulos) [inline packages/insomnia-app/app/ui/components/wrapper-debug.tsx:17]: :heart_decoration:

File: packages/insomnia-app/app/ui/components/wrapper-debug.tsx [hunk 4]
```diff
@@ -165,16 +160,8 @@ class WrapperDebug extends PureComponent<Props> {
...
-          moveDoc={handleMoveDoc}
-          // @ts-expect-error -- TSCONVERSION this prop isn't sent
-          hidden={sidebarHidden}
-          width={sidebarWidth}
-          workspace={activeWorkspace}
-          activeRequest={activeRequest}
           filter={sidebarFilter || ''}
           hotKeyRegistry={settings.hotKeyRegistry}
-          activeEnvironment={activeEnvironment}
-          activeSpace={activeSpace}
```
Comment (dimitropoulos) [inline packages/insomnia-app/app/ui/components/wrapper-debug.tsx:177]: so fulfilling

File: packages/insomnia-smoke-test/README.md [hunk 5]
```diff
@@ -81,15 +81,15 @@ Sometimes selecting by a React component and props, directly from `app.client` i
...
+  const request = await app.client.react$('UnconnectedSidebarRequestRow', {
     props: { isActive: true, request: { name } },
   });
 
   await request.waitForDisplayed();
 };
 
 export const clickFolderByName = async (app, name) => {
-  const folder = await app.client.react$('SidebarRequestGroupRow', {
+  const folder = await app.client.react$('UnconnectedRequestGroupRow', {
```
Comment (dimitropoulos) [inline packages/insomnia-smoke-test/README.md]: Is this a typo? I don't see `UnconnectedRequestGroupRow` anywhere.  Only `UnconnectedSidebarRequestGroupRow`.

File: packages/insomnia-app/app/ui/components/sidebar/dnd.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+export const hoverHandleCreator = <Props extends Object>(): Required<DropTargetSpec<Props>>['hover'] =>
+  (_, monitor, component) => {
+    if (isAbove(monitor, component)) {
+      component.setDragDirection(1);
+    } else {
+      component.setDragDirection(-1);
+    }
+  };
+
+const moveDoc = async ({
```
Comment (dimitropoulos) [inline packages/insomnia-app/app/ui/components/sidebar/dnd.tsx:72]: thank you for removing the useless `_` prefix from before

**Review by dimitropoulos — APPROVED** [review]
aside from one thing I believe to be a typo which is in a markdown file this has been a thoroughly enjoyable experience to review.

File: packages/insomnia-app/app/ui/components/wrapper-debug.tsx (hunk 4 above)
Comment (develohpanda) [reply packages/insomnia-app/app/ui/components/wrapper-debug.tsx:177]: Y E S

File: packages/insomnia-smoke-test/README.md (hunk 5 above)
Comment (develohpanda) [reply packages/insomnia-smoke-test/README.md]:
```suggestion
  const folder = await app.client.react$('UnconnectedSidebarRequestGroupRow', {
```

**Pull request merged**